onnxruntime/core/providers/webnn/builders/helper.cc

```diff
@@ -145,46 +141,47 @@ bool IsSupportedDataType(const int32_t onnx_data_type, const emscripten::val& we
   if (it == onnx_to_webnn_data_type_map.end())
     return false;
 
-  std::string webnn_data_type = it->second;
+  const std::string_view webnn_data_type = it->second;
 
   // Check if WebNN supports the data type.
-  emscripten::val is_supported = webnn_supported_data_types.call<emscripten::val>("includes",
-                                                                                  emscripten::val(webnn_data_type));
+  emscripten::val is_supported =
+      webnn_supported_data_types.call<emscripten::val>("includes", emscripten::val(webnn_data_type.data()));
```

Comment:
`emscripten::val(webnn_data_type.data()`

Note `string_view`'s `data()` is not guaranteed too be null terminated, which is a capable advantage of `string_view` since you can take a slice of a much larger string without allocating a temporary copy, but it means you should call the overloads of functions that take a pointer and length, rather than just length.

Does `emscripten::val` have an overload like `emscripten::val(webnn_data_type.data(), webnn_data_type.size())`?

Reply:
> Note string_view's data() is not guaranteed too be null terminated, which is a capable advantage of string_view since you can take a slice of a much larger string without allocating a temporary copy, but it means you should call the overloads of functions that take a pointer and length, rather than just length.

Good point, although the `webnn_data_type` is already a constant `string_view` (should be constructed with a null terminated), I should use `std::string(webnn_data_type)` to avoid the risk.

> Does emscripten::val have an overload like emscripten::val(webnn_data_type.data(), webnn_data_type.size())?

No, it doesn't.